### PR TITLE
fix: Return error message when relationship is not accepted in Comple…

### LINKED ISSUE #537: "Complete task" not including all the necessary information when request id is not in accepted mode
## Description
As a user,when i try to complete a task I need to know what I did wrong and why what I expected did not happen.

I observed that when a request id is not accepted and a user tries to complete a task (which does not exist, as a task cannot be created) the message is about the wrong task, but it should show the initial problem too, that the relationship is not accepted.

**To Reproduce**
1. Send a relationship request from user1 to a user2
2. Do not accept that request from user2
3. Try to complete a task (the number will be random as a task cannot be created)
4. See the error message, about the task not existing, but not about the relationship not having been accepted

**Expected behavior**
The message should inform the user trying to complete the task that the relationship is not accepted too.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by StergiosSozos**
### Description
The mentorship relation state (accepted or not) is now checked when someone is trying to complete a task from a relationship that is not accepted. This check is also of higher importance than the existence of the task. So if the relationship is not accepted the user gets a message that the relationship is not accepted.

Fixes #537

### Type of Change:
- Code

### How Has This Been Tested?
All the tests have the same result as before. No extra tests were added.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Review by atm1504 — DISMISSED** [review]

File: app/api/resources/mentorship_relation.py [hunk 1]
```diff
@@ -657,7 +657,14 @@ class UpdateTask(Resource):
     @mentorship_relation_ns.response(
         HTTPStatus.OK, "%s" % messages.TASK_WAS_ACHIEVED_SUCCESSFULLY
     )
-    @mentorship_relation_ns.response(HTTPStatus.BAD_REQUEST, "%s" % messages.TASK_WAS_ALREADY_ACHIEVED)
+    @mentorship_relation_ns.response(
+        HTTPStatus.BAD_REQUEST,
+        "%s\n%s"
```
Comment (isabelcosta) [inline app/api/resources/mentorship_relation.py]: Could you use f-strings, as per these [coding standards](https://github.com/anitab-org/mentorship-backend/wiki/Coding-Standards)

File: app/api/dao/task.py [hunk 2]
```diff
@@ -136,7 +136,8 @@ def complete_task(user_id: int, mentorship_relation_id: int, task_id: int):
 
         if not (user_id == relation.mentee_id or user_id == relation.mentor_id):
             return messages.USER_NOT_INVOLVED_IN_THIS_MENTOR_RELATION, HTTPStatus.UNAUTHORIZED
-
+        if relation.state != MentorshipRelationState.ACCEPTED:
+            return messages.UNACCEPTED_STATE_RELATION, HTTPStatus.BAD_REQUEST
```
Comment (isabelcosta) [inline app/api/dao/task.py:142]: add a blank line below please, just to separate visually each validation section

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@StergiosSozos I requested some minor changes. this is almost done, thank you for this contribution 🎉

File: app/api/resources/mentorship_relation.py (hunk 1 above)
Comment (StergiosSozos) [reply app/api/resources/mentorship_relation.py]: I changed it to f-strings!

File: app/api/dao/task.py (hunk 2 above)
Comment (StergiosSozos) [reply app/api/dao/task.py:142]: I added the blank spaces!

File: app/api/dao/task.py [hunk 3]
```diff
@@ -131,13 +131,17 @@ def complete_task(user_id: int, mentorship_relation_id: int, task_id: int):
...
         if relation is None:
             return messages.MENTORSHIP_RELATION_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND
 
         if not (user_id == relation.mentee_id or user_id == relation.mentor_id):
             return messages.USER_NOT_INVOLVED_IN_THIS_MENTOR_RELATION, HTTPStatus.UNAUTHORIZED
 
+        if relation.state != MentorshipRelationState.ACCEPTED:
+            return messages.UNACCEPTED_STATE_RELATION, HTTPStatus.BAD_REQUEST
         task = relation.tasks_list.find_task_by_id(task_id)
+
```
Comment (isabelcosta) [inline app/api/dao/task.py]: blank line was in line 143, and not 144

Comment (StergiosSozos) [reply app/api/dao/task.py]: Sorry for that, is it now correct?

**Review by isabelcosta — CHANGES_REQUESTED** [review]
I just recalled now, can you create a simple test for it, showing that that is the first message that appears if the user attempts to create a task in a non accepted relation.
This would make this PR shine 🤩

**Comment by StergiosSozos** [comment]
Sure, so a unit test  right? So that we can test that the message that we want to be diaplayed is the one i added here and it does appear.

**Comment by isabelcosta** [comment]
> Sure, so a unit test right? So that we can test that the message that we want to be diaplayed is the one i added here and it does appear.

yes :)

**Comment by StergiosSozos** [comment]
A unit test for my method is now added, in addition with an extra user on the databse.

**Comment by devkapilbansal** [comment]
@StergiosSozos can you please update this PR

**Comment by isabelcosta** [comment]
@StergiosSozos thank you for working on this project 🙌🏾 From what I see, only the merge conflicts are left to approve this. However, I will close this PR due to lack of response. If you wish to work on it again and the issue is still available feel free to comment there :)

**Comment by mariejp** [comment]
Hi! What were the requested changes for this PR? I cannot seem to locate them anywhere.

**Pull request closed without merging**